msystems/views.py

```diff
@@ -90,12 +83,29 @@ def acs(request):
         return redirect(MsystemsConfig.base_login_redirect)
 
 
-# Saml have it's own csrf protection, django not needed
+# Saml have its own csrf protection, django not needed
 @csrf_exempt
 @require_POST
 def sls(request):
-    # TODO implement SLS
-    return HttpResponse("Ok")
+    auth = _build_auth(request)
+    auth.process_slo()
+    errors = auth.get_errors()
+
+    if not errors:
+        username = auth.get_nameid()
+        user_data = auth.get_attributes()
+
+        logger.debug("Logout attempt, username=%s, user_data=%s", username, user_data)
+
+        if 'RelayState' in request.POST and _validate_relay_state(request.POST['RelayState']):
+            return redirect(auth.redirect_to(request.POST['RelayState']))
+        else:
+            return redirect(MsystemsConfig.base_login_redirect)
+    else:
+        logger.error("Logout attempt failed: %s\n%s", str(
+            errors[-1]), auth.get_last_error_reason())
+        # TODO Add information about failed login attempt for the user
+        return redirect(MsystemsConfig.base_login_redirect)
```

Comment:
I'd suggest reversing the condition:
```suggestion
    if errors:
        logger.error("Logout attempt failed: %s\n%s", str(
            errors[-1]), auth.get_last_error_reason())
        # TODO Add information about failed login attempt for the user
        return redirect(MsystemsConfig.base_login_redirect)
    
          username = auth.get_nameid()
      user_data = auth.get_attributes()
    
  logger.debug("Logout attempt, username=%s, user_data=%s", username, user_data)
  if 'RelayState' in request.POST and _validate_relay_state(request.POST['RelayState']):
      return redirect(auth.redirect_to(request.POST['RelayState']))
  else:
      return redirect(MsystemsConfig.base_login_redirect)
```

Reply:
I removed the sls code (To be implemented in different ticket, also different approach), but i applied this to the acs endpoint